presto-bytecode/src/main/java/com/facebook/presto/bytecode/CompilationException.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.bytecode;
+
+import static java.util.Objects.requireNonNull;
+
+public class CompilationException
+        extends RuntimeException
+{
+    public CompilationException(String message, RuntimeException cause)
+    {
+        super(requireNonNull(message, "message is null"), requireNonNull(cause, "cause is null"));
```

Comment:
Remove these null checks, exceptions never do that. This also loses the real exception